pkg/placement/plugins/addon/addon.go

```diff
@@ -82,7 +82,7 @@ func (c *AddOn) Score(ctx context.Context, placement *clusterapiv1beta1.Placemen
 		// get AddOnPlacementScores CR with resourceName
 		addOnScores, err := c.handle.ScoreLister().AddOnPlacementScores(namespace).Get(c.resourceName)
 		if err != nil {
-			klog.Warningf("Getting AddOnPlacementScores failed: %s", err)
+			klog.FromContext(ctx).Info("Failed to get AddOnPlacementScores", "error", err)
```

Comment:
How about add the logger to interface [Score](https://github.com/open-cluster-management-io/ocm/blob/main/pkg/placement/plugins/interface.go#L55
)
```
type Prioritizer interface {
	Plugin

	// Score gives the score to a list of the clusters, it returns a map with the key as
	// the cluster name.
	Score(ctx context.Context, logger Logger, placement *clusterapiv1beta1.Placement, clusters []*clusterapiv1.ManagedCluster) (PluginScoreResult, *framework.Status)
}
```
Then only need to pass the logger to Score in https://github.com/open-cluster-management-io/ocm/blob/main/pkg/placement/controllers/scheduling/schedule.go#L231
```
		scoreResult, status := p.Score(ctx, logger, placement, filtered)
```
So that all the placement plugins can use the logger, and don't need to init it by plugin itself.